fix/shooting.lua

```diff
@@ -0,0 +1,34 @@
+-- Fixes shooting/throwing options stacking on each other, causing null pointers and crashes
+--@ enable = true
+--@ module = true
+
+local aim_projectile = df.global.game.main_interface.adventure.aim_projectile
+
+local overlay = require('plugins.overlay')
+
+FixShootingOverlay = defclass(FixShootingOverlay, overlay.OverlayWidget)
+FixShootingOverlay.ATTRS{
+    desc='Adds keywords to conversation entries.',
+    default_enabled=true,
+    viewscreens={
+        'dungeonmode/Inventory',
+        'dungeonmode/AimProjectile'
+    },
+    frame={w=0, h=0},
+    overlay_onupdate_max_freq_seconds=0,
+}
+
+function FixShootingOverlay:overlay_onupdate()
+    if aim_projectile.open == false then
+        aim_projectile.shooter_it = nil
+        aim_projectile.ammo_it = nil
+        aim_projectile.thrown_it = nil
+        aim_projectile.shooting = false
+    elseif aim_projectile.shooting then
+        aim_projectile.thrown_it = nil
+    end
+end
```

Comment:
can this be replaced with
```lua
function FixShootingOverlay:onInput()
    if aim_projectile.open == false then
        aim_projectile.shooter_it = nil
        aim_projectile.ammo_it = nil
        aim_projectile.thrown_it = nil
        aim_projectile.shooting = false
    elseif aim_projectile.shooting then
        aim_projectile.thrown_it = nil
    end
end
```
that would run just before a shooting action since it would run in response to both keystrokes and mouse clicks